Io c

IgorTeplycj opened the pull request.

IgorTeplych committed:
Add dir

IgorTeplych committed:
Add new files

IgorTeplych committed:
Добавил тесты

IgorTeplych committed:
create

IgorTeplych committed:
Исправил юнит тесты

IgorTeplych committed:
Добавил команду RotateAndModifVelosityCommand и тесты к ней. Добавил команду проверки уровня топлива и тесты к ней.

IgorTeplych committed:
Добавил IoC контейнер

IgorTeplych committed:
Добавил контейнер команд и команду модифицирующую контейнер команд.

IgorTeplych committed:
Доработал IoC. Добавил однопоточные тесты.

IgorTeplych committed:
Добавил скоупы

IgorTeplych committed:
Добавил тесты скоупов

etyumentcev commented on HM2/IoCs/IoC.cs:
```diff
@@ -0,0 +1,43 @@
+        static IDictionary<string, T> container = new Dictionary<string, T>();
+        public static T Resolve(string key, params object[] obj)
+        {
+            if (key.Contains("Registration"))
```
А что, если придется поменять реализацию IoC?

IgorTeplycj replied:
Если придется поменять строковые ключи в IoC, то придется модифицировать весь код, где вызывается Resolve(), Но лучше этого не делать. А в самом начале условиться, что строка "Registration" является ключом регистрации новой зависимости. Для добавления новых ключей, чтобы не писать множество проверок, можно использовать словарь, в котором ключ - строка, а значение ссылка на метод. Тогда добавляем в словарь новый ключ и ссылку на новый метод.

etyumentcev commented on HM2/IoCs/IoC.cs:
```diff
@@ -0,0 +1,43 @@
+            }
+        }
+
+        static T Add(string key, T val)
```
А если придется удалять зависимость, как это сделать без модификации класса?

IgorTeplycj replied:
Модифицировал код по первому вопросу. Так как мне кажется лучше.

IgorTeplycj replied:
Добавил тест удаления ссылочного типа из IoC. Для удаления объекта без модификации кода IoC перерегистрирую команду со значением null. При вызове такой команды теперь будет возвращаться null.

IgorTeplycj commented:
Тест

IgorTeplych committed:
Переработал IoC. Добавил таблицу методов.

IgorTeplych committed:
Добавил тест удаления объекта из IoC. Для удаления ссылочного типа нужно перерегистрировать ссылочный тип в контейнере со значением null.

IgorTeplych committed:
тест

IgorTeplych committed:
Добавил интерфейс IMovable

IgorTeplych committed:
Добавил адаптер IMovable, и тесты для него

IgorTeplych committed:
Добавил генератор кода

IgorTeplych committed:
Закоментировал метод генератора кода

IgorTeplych committed:
Исправил генератор кода

IgorTeplych committed:
Исправил генератор

IgorTeplych committed:
Добавил автогенератор и тесты сгенерированного адаптера.

IgorTeplych committed:
Откатил пакеты до версий
Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.8.0
Microsoft.CodeAnalysis.Analyzers" Version="3.3.1"
Теперь можно перейти по ссылке на сгенерированный код.

IgorTeplych committed:
Изменил
<TargetFramework>net5.0</TargetFramework>
на
<TargetFramework>net6.0</TargetFramework>

IgorTeplych committed:
откатил до net 5.0

IgorTeplych committed:
Изменил файл HM2.csproj

IgorTeplych committed:
-

IgorTeplych committed:
-

IgorTeplych committed:
-

IgorTeplych committed:
-

IgorTeplych committed:
Удалил генератор кода

IgorTeplych committed:
-

IgorTeplych committed:
Добавил генератор

IgorTeplych committed:
Добавил тесты сгенерированного кода адаптера

IgorTeplych committed:
Удалил адаптер, написанный вручную. Переработал генератор (теперь он генерирует методы, возвращающие void). Добавил в интерфейс IMovable метод void Finish();

IgorTeplych committed:
Добавил тест метода Finish() авто адаптера.
